src/algebra/pi_instances.lean

```diff
@@ -146,6 +146,9 @@ begin
   { intros x y, ext1 z, rw [is_ring_hom.map_add (f z)], refl, }
 end
 
+instance decidable_eq [fintype I] [Π i : I, decidable_eq (f i)] : decidable_eq (Π i : I, f i)
+  | f g := if h : _ then is_true $ funext h else is_false (mt congr_fun h)
+
```

Comment:
This is actually a duplicate of an instance in `data/fintype`, `fintype.decidable_pi_fintype`. 

I'm guessing your code doesn't work without it. Could you locally increase the priority of `fintype.decidable_pi_fintype` instead of this?

Reply:
It used to not work without that instance, but now it just compiled after I removed it. So I removed it and did not need to increase the priority.